packages/connectors/package.json

```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@oasisdex/connectors",
+  "version": "0.0.5",
+  "description": "Web3-react connectors for Ledger, trezor and magic link",
+  "homepage": "https://github.com/OasisDEX/common#readme",
+  "main": "lib/src/index.js",
+  "typings": "lib/src/index.d.ts",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/OasisDEX/common.git"
+  },
+  "scripts": {},
+  "bugs": {
+    "url": "https://github.com/OasisDEX/common/issues"
+  },
+  "devDependencies": {
+    "@types/ledgerhq__hw-transport-u2f": "^4.21.2",
+    "@types/lodash": "^4.14.164",
+    "@types/web3-provider-engine": "^14.0.0"
+  },
+  "dependencies": {
+    "@0x/assert": "^3.0.17",
+    "@0x/subproviders": "^6.1.9",
+    "@0x/utils": "^6.1.0",
+    "@ledgerhq/hw-app-eth": "^5.27.1",
+    "@ledgerhq/hw-transport-u2f": "^5.28.0",
+    "@web3-react/abstract-connector": "^6.0.7",
+    "@web3-react/types": "^6.0.7",
+    "ethereumjs-tx": "^2.1.2",
+    "ethereumjs-util": "^7.0.7",
+    "ethjs-util": "^0.1.6",
+    "hdkey": "^2.0.1",
+    "lodash": "^4.17.20",
```

Comment:
I am worried that we can easily end up bundling lodash twice if there is even a slight mismatch in versions here and in the consumer. We can actually measure if it's really a problem by measuring bundle size (for some projects we have even codechecks/bundlesize check configured). 

Anyway, I see the following solutions:
- keep common deps like lodash pinned to the same version across all projects
- make common deps peer dependencies
- relax version requirements

I think the last option is the easiest for now. If we make the version that we require as "broad" as possible ie something like `4.x` it should increase the chance of it being reused.

Reply:
This comment is probably misplaced as here `lodash` version is defined loosely and `^4.17.20` matches all the versions of the form `4.x.y && >=4.17.20`.

On the other hand - I used to have pinned versions in `@oasisdex/transactions` as I needed them to generate compatible `yarn.lock` and forgot to remove 👎 . I just did it with the following commit: 911e5f1.

I've also checked if we don't have different `lodash` versions in the two frontends after installing those packages and we do. I checked why and found this in `yarn.lock`
```
lodash@4.17.15, lodash@^4.17.12, lodash@^4.17.13, lodash@^4.17.14, lodash@^4.17.15, lodash@^4.2.1:
  version "4.17.15"
  
lodash@^4.0.0, lodash@^4.13.1, lodash@^4.16.6, lodash@^4.17.10, lodash@^4.17.4, lodash@^4.3.0, lodash@~4.17.10:
  version "4.17.10"
  
lodash@^4.17.11:
  version "4.17.11"
```

There are two interesting things about the above snippet.
1. Someone requires a pinned version, but this someone is not me. It's `cypress`
2. More important one. `4.17.15` satisfies all the constraints in all three sets, but for whatever reason - yarn does not resolve them into one common version. `yarn` does that for stability.  That's what `yarn.lock`s are for. But what if we do want to upgrade `lodash` and test the app after? Run `yarn upgrade`? Not so fast. This will upgrade the root version but will not touch the `lodash` in subdependencies. Well then? How? One option is to manually edit yarn.lock and merge the constraints. Not beautiful, but works and we will see the changes clearly in the diff. Second option is to remove `yarn.lock` and regenerate. But this is not an option on any reasonably sized project. Third option is to have some command like `yarn upgrade --deep`. Everyone has this problem and they're working on that. https://github.com/yarnpkg/yarn/issues/4986 https://github.com/yarnpkg/yarn/issues/2543 etc. Unfortunately - it's not there yet. Option number 4 is to use yarn resolutions to resolve subdependencies to the same versions. Again - not the best idea. The last option is to write a small dependency merger script. It will read the `yarn.lock` and find sets of constraints that have a common solution and merge them. Obvsly asking the user to confirm.

Either way - let's discuss it further here or offline tomorrow